hugo/assets/js/species_cards.js

```diff
@@ -0,0 +1,284 @@
+/*
+Handles searching/filtering, ordering and pagination of the species cards on the home page.
+This script works with 'index.html' inside the hugo/layouts folder.
+
+On load:
+1. Species attributes needed for searching fetched from the <template> cards built for each species.
+2. Determine if pagination exists (decided by Hugo based on number of species and 'maxSpeciesPerPage' in 'hugo/config.yaml').
+3. Run the render function to display the species cards that should be shown.
+
+Once loaded:
+- Event listners created for search box, dropdown and pagination (if pagination exists).
+- Each event listner calls the main render function (renderSpeciesCards) alongside some others dependant on the event.
+
+Notes:
+- This code can handle both pagination and no pagination (i.e. not enough species to warrant it). So can the Playwright tests assocaited with this page.
+*/
+
+const SORT_OPTIONS = {
+    LAST_UPDATED: 'lastUpdated',
+    A_TO_Z: 'alphabet',
+    Z_TO_A: 'revAlphabet'
+};
+
+const paginationItems = document.querySelectorAll('.pagination .page-item');
+const noResultsCard = document.getElementById('no-filtered-card');
+const PAGINATION_EXISTS = paginationItems.length > 0; // if not enough species, there will be no pagination yet.
+const CARDS_PER_PAGE = parseInt(document.getElementById('card-container').dataset.numbCardsPerPage);
```

Comment:
A perhaps minor but general question on when to capitalize constants (not just `const`s but things that are not supposed to ever change) in JS. 

I feel like using all caps for: `SORT_OPTIONS` and `CARDS_PER_PAGE` makes sense but not sure if it would make sense to name `paginationItems` or `noResultsCard` which also never change (and should never change) with all caps. Do you have an opinion on that?

Reply:
Thanks for the tip! I agree it is good to be consistent and I think also predictable with naming so I have updated to only have `SORT_OPTIONS` as capitalized. commit: 8f71dba424b93c7fc2dfba1621dd5ec68a081672